subiquity/server/controllers/cmdlist.py

```diff
@@ -25,6 +27,22 @@
 from subiquity.server.controller import NonInteractiveController
 
 
+@attr.s(auto_attribs=True)
+class Command:
+    args: Union[str, Sequence[str]]
+    check: bool
+
+    def desc(self) -> str:
```

Comment:
I'm fine with this, just curious - was the built-in representation not great?

Reply:
I think it's just that this way of formatting is what is associated with the "start run" event. It's visible to the user iirc, so probably should stay friendly (I wonder if we should run it though shlex.quote though?)

Reply:
So, the installation screen displays the commands that are run. When commands are specified as an array of arguments e.g., `["ls", "-l", "/"]`, they are transformed into a string to be more human readable.

I did not change the existing behavior here but moved the implementation in the Command object so we don't need that many branches in the the `run` function.

```diff
-            if isinstance(cmd, str):
-                desc = cmd
-            else:
-                desc = ' '.join(cmd)
+            desc = cmd.desc()
```

I will add some docstrings to make this clearer.

Reply:
> (I wonder if we should run it though shlex.quote though?)

sure, let's `shlex.join` this. That will allow users/us to copy paste without losing information.